catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/stage/PlaySoundBrickStageTest.java

```diff
@@ -0,0 +1,129 @@
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.stage;
+
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.SoundInfo;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.PlaySoundBrick;
+import org.catrobat.catroid.io.SoundManager;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.FileTestUtils;
+import org.catrobat.catroid.uiespresso.util.hardware.SensorTestArduinoServerConnection;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.util.List;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+@RunWith(AndroidJUnit4.class)
+public class PlaySoundBrickStageTest {
+	private static final int RESOURCE_SOUND = org.catrobat.catroid.test.R.raw.longsound;
+	private String soundName = "testSound1";
+	private File soundFile;
+	private List<SoundInfo> soundInfoList;
+	private int waitingTime = 2500;
+
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> programMenuActivityRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject();
+		programMenuActivityRule.launchActivity(null);
+	}
+
+	@After
+	public void tearDown() {
+		if (soundFile != null && soundFile.exists()) {
+			soundFile.delete();
+		}
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Functional.class, Cat.Gadgets.class, Cat.SettingsAndPermissions
+			.class, Cat.SensorBox.class})
+	@Test
+	public void testSoundPlayedFromPlaySoundBrick() {
+		SensorTestArduinoServerConnection.checkAudioSensorValue(SensorTestArduinoServerConnection
+				.SET_AUDIO_OFF_VALUE, waitingTime);
+
+		onView(withId(R.id.button_play))
+				.perform(click());
+
+		SensorTestArduinoServerConnection.checkAudioSensorValue(SensorTestArduinoServerConnection
+				.SET_AUDIO_ON_VALUE, waitingTime);
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Functional.class, Cat.Gadgets.class, Cat.SettingsAndPermissions
+			.class, Cat.SensorBox.class})
+	@Test
+	public void testNoSoundPlayed() {
+		if (soundFile != null && soundFile.exists()) {
```

Comment:
is this really supposed to be working like this?
deleting the file, while still leaving the reference in soundInfoList and the brick.

if this is like the name suggests just supposed to just stay quite, use another create project that doenst add a sound in the first place and move the createProject and launchActivity in the respecive tests

Reply:
TY for the fast review! I pushed the refactored version.